File: src/nativeMain/kotlin/state/GameState.kt
package com.a_dinosaur.kotlinsdl.state

import com.a_dinosaur.kotlinsdl.*

class GameState: State
{
	val guy = LittleGuy()

	override fun tick(deltaTime: Float)
	{
		guy.update()
	}

	override fun draw(deltaTime: Float)
	{
		Renderer.setDrawColour(Renderer.Colour(0x1F1F1FFFu))
		Renderer.clear()

		guy.draw()
	}
}

File: src/state/gamestate.rs
use crate::actor::Actor;
use crate::actor::beato::Beato;
use crate::renderer::Renderer;
use crate::state::{State, StateCmd};

#[derive(Default)]
pub(crate) struct GameState {
  beato: Beato
}

impl State for GameState {
  fn tick(&mut self, deltatime: f32) -> StateCmd {
    self.beato.update(deltatime);
    StateCmd::Continue
  }

  fn load(&mut self, renderer: &mut Renderer) {
    self.beato.load_textures(renderer);
  }

  fn draw(&mut self, renderer: &mut Renderer, deltatime: f32) {
    renderer.clear_colour(0x1F, 0x1F, 0x1F);
    self.beato.draw(renderer, deltatime);
  }
}
